File: src/main/java/com/yanghui/LingYueBot/core/coreTools/ParseOperationCode.java
package com.yanghui.LingYueBot.core.coreTools;

public class ParseOperationCode {

    public static int parseOperationCode(String operation) {
        String[] instructionList = operation.split(" ");
        switch (instructionList[0]) {
            case "Like":
                return 0;
            case "Get":
                switch (instructionList[1]) {
                    case "-Like":
                        return 10;
                }
                break;
            case "DriftBottle":
                switch (instructionList[1]) {
                    case "-GET":
                        if (instructionList[2].equals("Local"))
                            return 20;
                        else if (instructionList[2].equals("Global"))
                            return 21;

                        break;
                    case "-ADD":
                        if (instructionList[2].equals("Local")) {
                            return 22;
                        } else if (instructionList[2].equals("Global")) {
                            return 23;
                        }
                        break;
                    case "-Like":
                        return 24;
                }
                break;
            case "Balance":
                return 30;
            case "RandCard":
                if (instructionList[1].equals("Normal"))
                    return 40;
                else if (instructionList[1].equals("Special"))
                    return 41;
            case "Satellite":
                return 50;
            case "MoePic":
                return 60;
            case "RandSeed":
                return 70;
            default:
                return -1;
        }
        return -1;
    }

}

File: src/main/java/com/yanghui/lingYueBot/core/coreTools/ParseOperationCode.kt
package com.yanghui.lingYueBot.core.coreTools

class ParseOperationCode {

    companion object {
        fun parseOperationCode(operation: String): Int {
            val instructionList: List<String> = operation.split(" ")
            return when (instructionList[0]) {
                "Like" -> 0
                "Get" -> when (instructionList[1]) {
                    "-Like" -> 10
                    else -> -1
                }
                "DriftBottle" -> when (instructionList[1]) {
                    "-GET" -> when (instructionList[2]) {
                        "Local" -> 20
                        "Global" -> 21
                        else -> -1
                    }
                    "-ADD" -> when (instructionList[2]) {
                        "Local" -> 22
                        "Global" -> 23
                        else -> -1
                    }
                    "-Like" -> 24
                    else -> -1
                }
                "Balance" -> 30
                "RandCard" -> when (instructionList[1]) {
                    "Normal" -> 40
                    "Special" -> 41
                    else -> 42
                }
                "Satellite" -> 50
                "MoePic" -> 60
                "RandSeed" -> 70
                else -> -1
            }
        }
    }

}